### PR TITLE
ARSN-185 Type Check patches/locationConstraints

### PR DESCRIPTION
**Pull request by ghivert**

### CODE REVIEW

**Comment by ghivert** [comment]
Bloop

**Review by miniscruff — APPROVED** [review]

File: lib/patches/locationConstraints.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                locationType: l.locationType,
+                sizeLimitGB: l.sizeLimitGB || null,
+                isTransient: Boolean(l.isTransient),
+                legacyAwsBehavior: Boolean(l.legacyAwsBehavior),
+            };
+            let supportsVersioning = false;
+            let pathStyle = process.env.CI_CEPH !== undefined;
+
+            switch (l.locationType) {
+            case 'location-mem-v1':
```
Comment (jbertran) [inline lib/patches/locationConstraints.ts]: Can we make `locationType` a literal union?

File: lib/patches/locationConstraints.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                location.type = 'mem';
```
Comment (jbertran) [inline lib/patches/locationConstraints.ts]: Ditto for `type` I guess

Comment (jbertran) [inline lib/patches/locationConstraints.ts:52]: Ditto for `type` I guess

**Review by jbertran — APPROVED** [review]

**Pull request merged**